language/move-lang/functional-tests/tests/diem/roles/basic.move

```diff
@@ -2,230 +2,243 @@
-        @{{bob}},
-        {{bob::auth_key}},
-        b"bob",
-        false
-    );
-}
-}
-// check: "Keep(ABORTED { code: 258,"
-
-//! new-transaction
-//! sender: blessed
-script {
-use DiemFramework::DiemAccount;
-fun main(account: signer) {
-    let account = &account;
-    DiemAccount::create_validator_account(
-        account,
-        @{{bob}},
-        {{bob::auth_key}},
-        b"bob",
-    );
-}
-}
-// check: "Keep(ABORTED { code: 2,"
-
-//! new-transaction
-//! sender: diemroot
-script {
-use DiemFramework::Roles;
-fun main(account: signer) {
-    let account = &account;
-    Roles::new_validator_role(account, account);
-}
-}
-// check: "Keep(ABORTED { code: 6,"
-
-//! new-transaction
-//! sender: blessed
-script {
-use DiemFramework::DiemAccount;
-fun main(account: signer) {
-    let account = &account;
-    DiemAccount::create_validator_operator_account(
-        account,
-        @{{bob}},
-        {{bob::auth_key}},
-        b"bob"
-    );
-}
-}
-// check: "Keep(ABORTED { code: 2,"
-
-//! new-transaction
-//! sender: diemroot
-script {
-use DiemFramework::Roles;
-fun main(account: signer) {
-    let account = &account;
-    Roles::new_validator_operator_role(account, account);
-}
-}
-// check: "Keep(ABORTED { code: 6,"
-
-//! new-transaction
-//! sender: diemroot
-script {
-use DiemFramework::DiemAccount;
-use DiemFramework::XUS::XUS;
-fun main(account: signer) {
-    let account = &account;
-    DiemAccount::create_parent_vasp_account<XUS>(
-        account,
-        @{{bob}},
-        {{bob::auth_key}},
-        b"bob",
-        false
-    );
-}
-}
-// check: "Keep(ABORTED { code: 258,"
-
-//! new-transaction
-//! sender: blessed
-script {
-use DiemFramework::Roles;
-fun main(account: signer) {
-    let account = &account;
-    Roles::new_parent_vasp_role(account, account);
-}
-}
-// check: "Keep(ABORTED { code: 6,"
-
-//! new-transaction
-//! sender: blessed
-script {
-use DiemFramework::DiemAccount;
-use DiemFramework::XUS::XUS;
-fun main(account: signer) {
-    let account = &account;
-    DiemAccount::create_child_vasp_account<XUS>(
-        account,
-        @{{bob}},
-        {{bob::auth_key}},
-        false
-    );
-}
-}
-// check: "Keep(ABORTED { code: 771,"
-
-//! new-transaction
-//! sender: blessed
-//! type-args: 0x1::XUS::XUS
-//! args: 0, {{vasp}}, {{vasp::auth_key}}, b"vasp", true
-stdlib_script::AccountCreationScripts::create_parent_vasp_account
-// check: "Keep(EXECUTED)"
-
-//! new-transaction
-//! sender: vasp
-script {
-use DiemFramework::Roles;
-fun main(account: signer) {
-    let account = &account;
-    Roles::new_child_vasp_role(account, account);
-}
-}
-// check: "Keep(ABORTED { code: 6,"
-
-//! new-transaction
-//! sender: diemroot
-script {
-use DiemFramework::Roles;
-fun main(account: signer) {
-    let account = &account;
-    assert(!Roles::can_hold_balance(account), 1);
-}
-}
-// check: "Keep(EXECUTED)"
+// This is only here because cargo test complains about empty transactions!
```

Comment:
I guess this is a test case that should be converted to using the move unit-test framework.

Reply:
Tim will probably volunteer to fix this.